diff --git a/builtin/gc.c b/builtin/gc.c
index d187cec1ea..09c019e5c5 100644
--- a/builtin/gc.c
+++ b/builtin/gc.c
@@ -2315,6 +2315,19 @@ static int is_systemd_timer_available(void)
 	return real_is_systemd_timer_available();
 }
 
+static int systemd_daemon_reload(void)
+{
+	struct child_process child = CHILD_PROCESS_INIT;
+
+	strvec_pushl(&child.args, "systemctl", "--user", "daemon-reload", NULL);
+
+	if (start_command(&child))
+		return error(_("failed to start systemctl"));
+	if (finish_command(&child))
+		return error(_("failed to run systemctl"));
+	return 0;
+}
+
 static char *xdg_config_home_systemd(const char *filename)
 {
 	return xdg_config_home_for("systemd/user", filename);
@@ -2439,8 +2452,28 @@ static int systemd_timer_write_service_template(const char *exec_path)
 	FILE *file;
 	const char *unit;
 	char *local_service_name = xstrfmt(SYSTEMD_UNIT_FORMAT, "", "service");
+	bool global_service;
+	struct stat st;
+
+	git_config_get_bool("maintenance.global-systemd-service",
+			    &global_service);
 
 	filename = xdg_config_home_systemd(local_service_name);
+	if (global_service) {
+		// ensure global servce file exists
+		if (stat("/etc/systemd/user/git-maintenance@.service", &st)) {
+			die(_("global git-maintenance@.service not found"));
+		}
+		res = 0;
+		// remove user servce file if it exists
+		if (unlink(filename)) {
+			if (!is_missing_file_error(errno))
+				warning(_("failed to delete '%s'"), filename);
+			goto error;
+		}
+		res = systemd_daemon_reload();
+		goto error;
+	}
 	if (safe_create_leading_directories(filename)) {
 		error(_("failed to create directories for '%s'"), filename);
 		goto error;
